yb-voyager/src/tgtdb/oracle.go

```diff
@@ -625,8 +653,9 @@ func (tdb *TargetOracleDB) IfRequiredQuoteColumnNames(tableName string, columns
 }
 
 // execute all events sequentially one by one in a single transaction
-func (tdb *TargetOracleDB) ExecuteBatch(migrationUUID uuid.UUID, batch EventBatch) error {
+func (tdb *TargetOracleDB) ExecuteBatch(migrationUUID uuid.UUID, batch *EventBatch) error {
 	// TODO: figure out how to avoid round trips to Oracle DB
+	batch.EventCountsByTable = batch.GetEventCountsByTable(tdb.tconf.Schema)
```

Comment:
Generally, you should update fields of a class only from its methods.
If you implement the `NewEventBatch()` as suggested above, you won't need this statement.